src/geometry/manifold/charted_space.lean

```diff
@@ -364,6 +364,10 @@ class closed_under_restriction (G : structure_groupoid H) : Prop :=
 (closed_under_restriction : ∀ {e : local_homeomorph H H}, e ∈ G → ∀ (s : set H), is_open s →
   (e : local_homeomorph H H).restr s ∈ G)
 
+lemma closed_under_restriction' (G : structure_groupoid H) [h : closed_under_restriction G] :
+  ∀ {e : local_homeomorph H H}, e ∈ G → ∀ (s : set H), is_open s →
+  (e : local_homeomorph H H).restr s ∈ G := h.closed_under_restriction
```

Comment:
Instead of stating this with foralls, can you write it with assumptions left of the colon? This is more readable in general. Also, the proof should be on its own line. And `G` can be implicit as it appears in a further argument (and in the conclusion)

Reply:
I made all these changes, and made `s` implicit as well.  If that was taking the principle too far, let me know and I'll change it back in a second PR :)